### ISSUE TITLE
L515 capture error when built against librealsense v2.41

### ISSUE THREAD

**Issue by AH50** [intel-isl/Open3D#2985]
**Describe the bug**
Hi, I just built the open3d library from source. I wanted to test recording and also scene reconstruction on the L515. When I run the script(source http://www.open3d.org/docs/latest/tutorial/sensor/realsense.html): 
bin/examples/RealSenseRecorder --config ../examples/test_data/rs_default_config.json --record test_data.bag
I get the RGB camera view + depth(and it is working normally). 

When I press spacebar to record the scene the image crashes and after a while it doesn't respond.
![image](https://user-images.githubusercontent.com/34249083/106742762-be6f9300-661d-11eb-9ab8-23b450a8c381.png)

I am able to press the "Esc" button and I get an empty bag file.

When I try to run the script again I get a message:
Open3D/cpp/open3d/t/io/sensor/realsense/RealSenseSensor.cpp:209 [Open3D Error] Invalid RealSense camera configuration, or camera not connected:
UNKNOWN: No device connected

Sometimes it helps to unplug the device and plug it back in but sometimes I have to restart the computer.
I also tried to change configuration in /examples/test_data/rs_default_config.json file but it did not help.

rs_default_config.json(without any changes):
{
    "serial": "",
    "color_format": "RS2_FORMAT_ANY",
    "color_resolution": "0,0",
    "depth_format": "RS2_FORMAT_ANY",
    "depth_resolution": "0,0",
    "fps": "0",
    "visual_preset": ""
}

I also tried to use the script that is on the website but when I run it nothing happens. 
Code from website:
```bash
import json
import open3d as o3d
with open(config_filename) as cf:
    rs_cfg = o3d.t.io.RealSenseSensorConfig(json.load(cf))

rs = o3d.t.io.RealSenseSensor()
rs.init_sensor(rs_cfg, 0, bag_filename)
rs.start_capture(True)  # true: start recording with capture
for fid in range(150):
    im_rgbd = rs.capture_frame(True, True)  # wait for frames and align them
    # process im_rgbd.depth and im_rgbd.color

rs.stop_capture()
```

**To Reproduce**

1. run script : bin/examples/RealSenseRecorder --config ../examples/test_data/rs_default_config.json --record test_data.bag

**Expected behavior**
I should get the bag file that is needed to reconstruct the scene

**Environment:**
- Operating System: Ubuntu 20.04.2 LTS
- Kernel: Linux 5.8.0-41-generic
- Architecture: x86-64
- Python version: Python 3.7.6
- Open3D version: newest 0.12.0
- How did you install Open3D?: build from source
- Compiler version (if built from source): gcc 9.3.0

**Comment by ssheorey** [comment]
Hi @AH50 thanks for reporting this:

> When I press spacebar to record the scene the image crashes and after a while it doesn't respond.

Can you ensure nothing else is accessing the camera at the same time? It's probably a good idea to close `realsense-viewer` before using any other code to access the camera (it looks open in the screenshot).

> UNKNOWN: No device connected

This can happen if the camera was not closed properly, e.g. by a an earlier crash. As you've noted, re-plugging the camera should fix this. 

> I also tried to use the script that is on the website but when I run it nothing happens.

This is sample code with placeholders:
`# process im_rgbd.depth and im_rgbd.color`
so just running this will not output or show any thing. You can add operations to the captured frames here as you like. e.g.:
`print(im_rgbd)`
Should print out some basic image properties (size and data type of the captured images).

**Comment by AH50** [comment]
Hi @ssheorey,
I tested many options and also like realsense-viewer was disabled.
After running this script:
```bash
import json
import open3d as o3d
with open("cf") as cf:
    rs_cfg = o3d.t.io.RealSenseSensorConfig(json.load(cf))

rs = o3d.t.io.RealSenseSensor()
rs.init_sensor(rs_cfg, 0, "tmpx.bag")
rs.start_capture(True)  # true: start recording with capture
for fid in range(3):
    im_rgbd = rs.capture_frame(True, True)  # wait for frames and align them
    # process im_rgbd.depth and im_rgbd.color
    print(im_rgbd)

rs.stop_capture()
```
(Recording time about 2 minutes)

I get: 
![image](https://user-images.githubusercontent.com/34249083/107336886-f2cdce00-6ab9-11eb-818d-f2745548067e.png)

One frame records for about 20 seconds but is blank.

**Comment by ssheorey** [comment]
I was able to reproduce this in the case that:
`USE_SYSTEM_LIBREALSENSE=ON`
and the externally installed (system) librealsense version is 2.41. Something may have broken from 2.40 -> 2.41.
Can you confirm this is the case for you? If so, you should be able to fix this by setting `USE_SYSTEM_LIBREALSENSE=OFF` and rebuilding Open3D.

**Comment by AH50** [comment]
I rebuilt open3d with the flag ``USE_SYSTEM_LIBREALSENSE=OFF`` and now I can record normally. Thank you.

Another problem has arisen.I tried to reconstruct the scene from my recording and got only one RGB and depth photo each and the *bag file itself weighs 900MB(recording 4-5 seconds).The RGB photo is completely black. The depth photo seems to be correct. It created one "fragment" and "scene" from these two photos. 

I also tried to reconstruct the *bag file which is given on the site and I had no problems with it. The scene built normally from the L515_test.bag file which weighed 37MB. I noticed that there is a difference in resolution because in the L515_test file it is using 960x540 and in my case it is 1280x720.  I didn't edit configuration file of the recording script, I used examples following the guidelines on open3d(http://www.open3d.org/docs/latest/tutorial/sensor/realsense.html) website

For recording I used:
 ``
bin/examples/RealSenseRecorder --config ../examples/test_data/rs_default_config.json --record test_data.bag
``

For the reconstruction I used:
 ``
python run_system.py config/realsense.json --make --register --refine --integrate
``


This is my realsense.json file:

{
	"name": "realsense bag file",
	"path_dataset": "/home/user/scans/L515_test.bag",
	"max_depth": 3.0,
	"voxel_size": 0.05,
	"depth_scale" : 3999.999755859375,
	"max_depth_diff": 0.07,
	"preference_loop_closure_odometry": 0.1,
	"preference_loop_closure_registration": 5.0,
	"tsdf_cubic_size": 3.0,
	"icp_method": "color",
	"global_registration": "ransac",
	"python_multi_threading": true
}
